### PR TITLE
Schedule special coverage

### PR DESCRIPTION
**Pull request by benghaziboy**
@mparent61 @MichaelButkovic

### CODE REVIEW

File: bulbs/utils/methods.py [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+    utc_now = timezone.now()
+    central_tz = tz.gettz('America/Chicago')
+    central_now = utc_now.astimezone(central_tz)
+    return central_now
+
+
+def today():
+    # catching this because settings.TODAY might literally be None in tests
+    # getattr will return None if the value is set to None
+    _today = getattr(settings, "TODAY", None)
```
Comment (mparent61) [inline bulbs/utils/methods.py]: 1-liner: 

return getattr(settings, "TODAY", get_central_now().date())

File: bulbs/special_coverage/utils.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from .models import SpecialCoverage
+
+
+def get_month_start_date():
+    """Returns the first day of the current month"""
+    now = get_central_now()
+    return timezone.datetime(day=1, month=now.month, year=now.year, tzinfo=now.tzinfo)
+
+
+def get_absurd_end_date():
```
Comment (mparent61) [inline bulbs/special_coverage/utils.py]: hahaha

File: bulbs/special_coverage/models.py [hunk 3]
```diff
@@ -43,17 +47,31 @@ def clean(self):
...
         super(SpecialCoverage, self).save(*args, **kwargs)
 
         if self.query and self.query != {}:
-            if self.active:
+            if self.is_active:
                 self._save_percolator()
             else:
                 self._delete_percolator()
 
+    def validate_publish_dates(self):
```
Comment (mparent61) [inline bulbs/special_coverage/models.py]: nice

File: bulbs/special_coverage/models.py [hunk 4]
```diff
@@ -100,6 +118,14 @@ def get_content(self, published=True):
         return search
 
     @property
+    def is_active(self):
+        now = get_central_now()
```
Comment (mparent61) [inline bulbs/special_coverage/models.py]: shouldn't this be "today()" to allow settings file override?

File: bulbs/special_coverage/management/commands/migrate_active_to_published.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from django.core.management.base import BaseCommand
+
+from ...utils import publish_active_special_coverages
+
+
+class Command(BaseCommand):
+    help = "Migrates Special Coverages to published"
+
+    def handle(self, *args, **kwargs):
+        publish_active_special_coverages()
```
Comment (mparent61) [inline bulbs/special_coverage/management/commands/migrate_active_to_published.py]: any reason this is in a separate module? I can't see this being used outside of the 1-time migration command

Comment (benghaziboy) [inline bulbs/special_coverage/management/commands/migrate_active_to_published.py]: fair.

**Comment by benghaziboy** [comment]
@mparent61 updated

**Comment by MichaelButkovic** [comment]
:+1:

**Pull request merged**